### ISSUE TITLE
Template literal parsing error

### ISSUE THREAD

**Issue by zpao** [jquery/esprima#1143]
There appear to be some cases that broke with recent changes.

This fails to parse:
```js
if (condition) {
  (`${foo}`)
}
```

However all of these do parse:

```js
if (condition) {
  `${foo}`
}
```
```js
if (condition)
  `${foo}`
```
```js
if (condition)
  (`${foo}`)
```

Namely there seems to be a problem with the interaction of parens & braces with template literals, probably due to #1104.

cc @mikesherov, and @nzakas (espree is failing in the same way)

**Comment by mikesherov** [comment]
@zpao investigating...

**Comment by ikarienator** [comment]
@mikesherov I can take this if you are too busy.

**Comment by mikesherov** [comment]
I'm on it. I have the answer.

**Comment by mikesherov** [comment]
Basically, `lookahead2` lexs 2 tokens out in harmony, something we forbid in master. So when I ported this to master, everything is fine. Now, for harmony, I have to basically only push onto the curlyStack when I haven't backtracked :-\

Fix is fairly straightforward, but I'm glad we don't have backtracking in master. Fix incoming.

**Comment by mikesherov** [comment]
Fixed by #1145
